GMX-Bots/gmx7-detect-profitable-trades/src/agent.spec.ts

```diff
@@ -0,0 +1,448 @@
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      });
+    let findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+
+    expect(findings).toStrictEqual([
+      Finding.fromObject({
+        name: "Unusual amount of profitable trades",
+        description: `User ${createAddress("0xf0").toLowerCase()} has a ${(15 / 16) * 100}% profitable trade ratio`,
+        alertId: "GMX-07",
+        protocol: "GMX",
+        severity: FindingSeverity.Medium,
+        type: FindingType.Suspicious,
+        metadata: {
+          account: createAddress("0xf0").toLowerCase(),
+          profitableTrades: "15",
+          totalTrades: "16",
+          totalProfit: "298985", //in USD
+        },
+      }),
+    ]);
```

Comment:
Similar to the other unit test, we should check with `expect().toStrictEqual()` in between the calls to the handler then `expect` an actual finding when it is supposed to be alerted.

For example
```
    let findings = await handler(transaction);
    expect(findings).toStrictEqual([]);

    findings = await handler(transaction);
    expect(findings).toStrictEqual([]);

    findings = await handler(transaction);
    expect(findings).toStrictEqual([
        Finding.fromObject({...});
    ]);